compaction.go

```diff
@@ -153,6 +154,8 @@ func (k compactionKind) String() string {
 		return "elision-only"
 	case compactionKindRead:
 		return "read"
+	case compactionKindTombstoneDensity:
+		return "tombstone-density"
```

Comment:
I'm not very opinionated on this and am happy to have a new compaction kind too, but to me this seems more like a default compaction that's just scored differently than a different compaction type altogether. Most of the compaction types here just run very differently; eg. elision-only rewrites L6 files, rewrite rewrites a file wherever it is, copy does a byte-by-byte copy, etc. So maybe we should fold this with default?

The other bonus of removing this compaction type would be it'd significantly shrink this PR as a lot of the changes are just about propagating the addition of the compaction type through metrics etc.